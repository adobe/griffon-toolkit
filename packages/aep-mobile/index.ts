// @create-index

export { default as analyticsResponse } from './src/analyticsResponse.js';
export { default as clientInfoAndroid } from './src/clientInfoAndroid.js';
export { default as clientInfoIOS } from './src/clientInfoIOS.js';
export { default as configUpdateCommand } from './src/configUpdateCommand.js';
export { default as configuration } from './src/configuration.js';
export { default as configurationUpdate } from './src/configurationUpdate.js';
export { default as edgeErrorResponse } from './src/edgeErrorResponse.js';
export { default as edgeRequest } from './src/edgeRequest.js';
export { default as edgeResponse } from './src/edgeResponse.js';
export { default as fakeEventCommand } from './src/fakeEventCommand.js';
export { default as genericPlaces } from './src/genericPlaces.js';
export { default as genericTrack } from './src/genericTrack.js';
export { default as identityResponse } from './src/identityResponse.js';
export { default as lifecycleStart } from './src/lifecycleStart.js';
export { default as locationAuthStatus } from './src/locationAuthStatus.js';
export { default as logForwardCommand } from './src/logForwardCommand.js';
export { default as mediaCreateTrackerRequest } from './src/mediaCreateTrackerRequest.js';
export { default as mediaSessionCreated } from './src/mediaSessionCreated.js';
export { default as mobileEvent } from './src/mobileEvent.js';
export { default as placesEntry } from './src/placesEntry.js';
export { default as placesExit } from './src/placesExit.js';
export { default as pushTrackingRequest } from './src/pushTrackingRequest.js';
export { default as receivePlaces } from './src/receivePlaces.js';
export { default as requestPlaces } from './src/requestPlaces.js';
export { default as screenshotCaptureCommand } from './src/screenshotCaptureCommand.js';
export { default as setPushIdentifier } from './src/setPushIdentifier.js';
export { default as sharedState } from './src/sharedState.js';
export { default as sharedStateConfig } from './src/sharedStateConfig.js';
export { default as sharedStateVersions } from './src/sharedStateVersions.js';
export { default as startEventForwardingCommand } from './src/startEventForwardingCommand.js';
export { default as startPlacesMonitor } from './src/startPlacesMonitor.js';
export { default as trackAction } from './src/trackAction.js';
export { default as trackMedia } from './src/trackMedia.js';
export { default as trackState } from './src/trackState.js';

