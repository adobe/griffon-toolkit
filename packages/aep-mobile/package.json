{
  "name": "@adobe/griffon-toolkit-aep-mobile",
  "description": "Events definitions for the AEP Mobile SDK",
  "version": "0.6.0",
  "repository": {
    "type": "git",
    "url": "https://github.com/adobe/griffon-toolkit.git",
    "directory": "packages/aep-mobile"
  },
  "publishConfig": {
    "access": "public"
  },
  "author": "Adobe Systems",
  "license": "Apache-2.0",
  "files": [
    "dist/"
  ],
  "dependencies": {
    "@adobe/griffon-toolkit": "*"
  },
  "scripts": {
    "prebuild": "rm -rf dist && mkdir dist",
    "build": "babel src --out-dir dist --root-mode upward"
  },
  "main": "dist/index.js"
}
