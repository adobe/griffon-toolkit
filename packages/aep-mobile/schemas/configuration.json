{
  "$id": "http://griffon.adobe.com/schemas/aep-mobile/configuration",
  "shortDesc": "Configuration Event",
  "group": "event",
  "allOf": [{ "$ref": "http://griffon.adobe.com/schemas/aep-mobile/mobileEvent" }],
  "type": "object",
  "properties": {
    "payload": {
      "inherit": true,
      "type": "object",
      "properties": {
        "ACPExtensionEventData": {
          "inherit": true,
          "type": "object",
          "alias": "eventData",
          "description": "The full list of current configuration values",
          "properties": {
            "build.environment": {
              "alias": "buildEnvironment",
              "description": "In the Launch UI, the type of environment this configuration was generated for",
              "type": "string",
              "mock": "dev"
            },
            "experienceCloud.org": {
              "alias": "experienceCloudOrg",
              "description": "The IMS Org that the mobile app's config was created by",
              "type": "string",
              "mock": "abc@AdobeOrg"
            },
            "property.id": {
              "alias": "launchPropertyId",
              "description": "The ID of the property inside launch",
              "type": "string",
              "mock": "abcd1234"
            },
            "rules.url": {
              "alias": "rulesUrl",
              "description": "The URL to download the rules configuration for the property",
              "type": "string",
              "mock": "http://assets.adobedtm.com/abc/abcdefg-development-rules.zip"
            }
          }
        },
        "ACPExtensionEventSource": {
          "inherit": true,
          "const": "com.adobe.eventSource.responseContent"
        },
        "ACPExtensionEventType": {
          "inherit": true,
          "const": "com.adobe.eventType.configuration"
        }
      },
      "required": [
        "ACPExtensionEventData"
      ]
    }
  }
}