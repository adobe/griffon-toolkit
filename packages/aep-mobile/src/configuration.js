"use strict";
/*
Copyright 2020 Adobe. All rights reserved.
This file is licensed to you under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License. You may obtain a copy
of the License at http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software distributed under
the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
OF ANY KIND, either express or implied. See the License for the specific language
governing permissions and limitations under the License.
*/
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
exports.__esModule = true;
var kit = __importStar(require("@adobe/griffon-toolkit"));
var configuration_json_1 = __importDefault(require("../schemas/configuration.json"));
/**
 * Contains constants and functions for a Configuration Event.
 *
 * The structure for a Configuration Event is as follows:
 * ```
 * {
 *   payload: {
 *     ACPExtensionEventData: {
 *       build.environment: <string>,
 *       experienceCloud.org: <string>,
 *       property.id: <string>,
 *       rules.url: <string>,
 *     },
 *     ACPExtensionEventSource: 'com.adobe.eventsource.responsecontent'
 *     ACPExtensionEventType: 'com.adobe.eventtype.configuration'
 *     ACPExtensionEventName: <string>,
 *     ACPExtensionEventNumber: <integer>,
 *     ACPExtensionEventUniqueIdentifier: <string>,
 *   },
 *   type: 'generic'
 *   vendor: <string>,
 *   annotations: <array>,
 *   clientId: <string>,
 *   timestamp: <number>,
 *   uuid: <string>,
 * }
 * ```
 *
 * @namespace configuration
 */
/**
 * Paths for the keys on a Configuration Event
 *
 * @enum {string}
 */
var path = {
    /** An object with custom data describing the event.<br />Path is `payload`. */
    payload: 'payload',
    /** The full list of current configuration values.<br />Path is `payload.ACPExtensionEventData`. */
    eventData: 'payload.ACPExtensionEventData',
    /** In the Launch UI, the type of environment this configuration was generated for.<br />Path is `payload.ACPExtensionEventData."build.environment"`. */
    buildEnvironment: 'payload.ACPExtensionEventData."build.environment"',
    /** The IMS Org that the mobile app's config was created by.<br />Path is `payload.ACPExtensionEventData."experienceCloud.org"`. */
    experienceCloudOrg: 'payload.ACPExtensionEventData."experienceCloud.org"',
    /** The ID of the property inside launch.<br />Path is `payload.ACPExtensionEventData."property.id"`. */
    launchPropertyId: 'payload.ACPExtensionEventData."property.id"',
    /** The URL to download the rules configuration for the property.<br />Path is `payload.ACPExtensionEventData."rules.url"`. */
    rulesUrl: 'payload.ACPExtensionEventData."rules.url"',
    /** The event source.<br />Path is `payload.ACPExtensionEventSource`. */
    eventSource: 'payload.ACPExtensionEventSource',
    /** The event type.<br />Path is `payload.ACPExtensionEventType`. */
    eventType: 'payload.ACPExtensionEventType',
    /** The name of the event.<br />Path is `payload.ACPExtensionEventName`. */
    eventName: 'payload.ACPExtensionEventName',
    /** The event number generated by the SDK.<br />Path is `payload.ACPExtensionEventNumber`. */
    sdkEventNumber: 'payload.ACPExtensionEventNumber',
    /** The unique event id.<br />Path is `payload.ACPExtensionEventUniqueIdentifier`. */
    eventId: 'payload.ACPExtensionEventUniqueIdentifier',
    /** The type of event.<br />Path is `type`. */
    rootType: 'type',
    /** The vendor of the plugin that sent the event.<br />Path is `vendor`. */
    vendor: 'vendor',
    /** Array of Annotation objects.<br />Path is `annotations`. */
    annotations: 'annotations',
    /** A unique id that differentiates clients from one another.<br />Path is `clientId`. */
    clientId: 'clientId',
    /** When the event occurred.<br />Path is `timestamp`. */
    timestamp: 'timestamp',
    /** Uniquely identifies each event.<br />Path is `uuid`. */
    rootId: 'uuid'
};
/**
 * Describes the number of parents this object has based off schema references. When checking for matches for example, we want to
 * use a schema that is more specific over a more generic schema
 *
 * @constant
 */
var parentDepth = 2;
/**
 * A label that can be used when describing this object
 */
var label = 'Configuration Event';
/**
 * A grouping for this object
 */
var group = 'event';
/**
 * The value for `eventSource` for a Configuration Event.
 *
 * Path is `payload,ACPExtensionEventSource`.
 *
 * @constant
 */
var EVENT_SOURCE = 'com.adobe.eventsource.responsecontent';
/**
 * The value for `eventType` for a Configuration Event.
 *
 * Path is `payload,ACPExtensionEventType`.
 *
 * @constant
 */
var EVENT_TYPE = 'com.adobe.eventtype.configuration';
/**
 * The value for `rootType` for a Configuration Event.
 *
 * Path is `type`.
 *
 * @constant
 */
var ROOT_TYPE = 'generic';
/**
 * Retrieves a value from the object. You can provide either a path or an alias.
 *
 * @function
 * @param {string} alias Path or alias
 * @param {*} data Data to search
 * @returns {*}
 */
var get = function (alias, data) {
    var func = function (data2) { return kit.search(path[alias] || alias, data2); };
    if (!data) {
        return func;
    }
    return func(data);
};
/**
 * Returns the `eventData` from the Configuration Event.
 * This is the the full list of current configuration values.
 *
 * Path is `payload,ACPExtensionEventData`.
 *
 * @function
 * @param {object} source The Configuration Event instance
 * @returns {object}
 */
var getEventData = kit.search(path.eventData);
/**
 * Returns the data using the specified path from the eventData
 * of the Configuration Event.
 *
 * @function
 * @param {...string} path key in object
 * @param {object} source The Configuration Event instance
 * @returns {*}
 */
var getEventDataKey = kit.curry(function (searchPath, source) { return kit.search(path.eventData + "." + searchPath, source); });
/**
 * Returns the `buildEnvironment` from the Configuration Event.
 * This is the in the Launch UI, the type of environment this configuration was generated for.
 *
 * Path is `payload,ACPExtensionEventData,build.environment`.
 *
 * @function
 * @param {object} source The Configuration Event instance
 * @returns {string}
 */
var getBuildEnvironment = kit.search(path.buildEnvironment);
/**
 * Returns the `experienceCloudOrg` from the Configuration Event.
 * This is the the IMS Org that the mobile app's config was created by.
 *
 * Path is `payload,ACPExtensionEventData,experienceCloud.org`.
 *
 * @function
 * @param {object} source The Configuration Event instance
 * @returns {string}
 */
var getExperienceCloudOrg = kit.search(path.experienceCloudOrg);
/**
 * Returns the `launchPropertyId` from the Configuration Event.
 * This is the the ID of the property inside launch.
 *
 * Path is `payload,ACPExtensionEventData,property.id`.
 *
 * @function
 * @param {object} source The Configuration Event instance
 * @returns {string}
 */
var getLaunchPropertyId = kit.search(path.launchPropertyId);
/**
 * Returns the `rulesUrl` from the Configuration Event.
 * This is the the URL to download the rules configuration for the property.
 *
 * Path is `payload,ACPExtensionEventData,rules.url`.
 *
 * @function
 * @param {object} source The Configuration Event instance
 * @returns {string}
 */
var getRulesUrl = kit.search(path.rulesUrl);
/**
 * Matcher can be used to find matching Configuration Event objects.
 *
 * @see kit.match
 * @constant
 */
var matcher = kit.combineAll([
    'payload.ACPExtensionEventSource==`com.adobe.eventsource.responsecontent`',
    'payload.ACPExtensionEventType==`com.adobe.eventtype.configuration`',
    'timestamp'
]);
/**
 * Tests the provided source against the matcher to see if it's Configuration Event event.
 *
 * @function
 * @param {object} source The Configuration Event instance
 * @returns {boolean}
 * @see kit.isMatch
 */
var isMatch = function (source) { return kit.isMatch(matcher, source); };
/**
 * Generates a Configuration Event with the const values set.
 * Can be useful in testing.
 * Can provide additional data by providing a flat object of paths and values.
 *
 * @function
 * @param {...Function} input Overrides
 * @returns {object}
 */
var make = function (input) { return kit.expandWithPaths(path, __assign({ eventSource: 'com.adobe.eventsource.responsecontent', eventType: 'com.adobe.eventtype.configuration', rootType: 'generic' }, input)); };
/**
 * Generates a Configuration Event with some default values set.
 * Can be useful in testing.
 * Can override defaults and provide additional data by providing a flat object
 * of paths and values.
 *
 * @function
 * @param {...Function} input Overrides
 * @returns {object}
 */
var mock = function (input) { return kit.expandWithPaths(path, __assign({ buildEnvironment: 'dev', experienceCloudOrg: 'abc@AdobeOrg', launchPropertyId: 'abcd1234', rulesUrl: 'http://assets.adobedtm.com/abc/abcdefg-development-rules.zip', eventSource: 'com.adobe.eventsource.responsecontent', eventType: 'com.adobe.eventtype.configuration', rootType: 'generic', vendor: 'com.adobe.mobile.sdk', clientId: 'appleABC', timestamp: Date.parse('12 Jan 2020 07:23:17 GMT'), rootId: '123' }, input)); };
/* ADD CUSTOM CONTENT BELOW */
// additional exports should be added here:
var customExports = {};
/* END CUSTOM CONTENT */
/* The content below is autogenerated. Do not make any changes */
exports["default"] = __assign(__assign({ path: path, mock: mock, make: make, schema: configuration_json_1["default"], get: get }, customExports), { getEventData: getEventData, getEventDataKey: getEventDataKey, getBuildEnvironment: getBuildEnvironment, getExperienceCloudOrg: getExperienceCloudOrg, getLaunchPropertyId: getLaunchPropertyId, getRulesUrl: getRulesUrl, isMatch: isMatch, matcher: matcher, EVENT_SOURCE: EVENT_SOURCE, EVENT_TYPE: EVENT_TYPE, ROOT_TYPE: ROOT_TYPE, label: label, group: group, parentDepth: parentDepth });
