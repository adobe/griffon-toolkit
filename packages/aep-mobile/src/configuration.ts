/*
Copyright 2020 Adobe. All rights reserved.
This file is licensed to you under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License. You may obtain a copy
of the License at http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software distributed under
the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
OF ANY KIND, either express or implied. See the License for the specific language
governing permissions and limitations under the License.
*/

import * as R from 'ramda';
import * as kit from '@adobe/griffon-toolkit';
import schema from '../schemas/configuration.json';

/**
 * Contains constants and functions for a Configuration Event.
 *
 * The structure for a Configuration Event is as follows:
 * ```
 * {
 *   payload: {
 *     ACPExtensionEventData: {
 *       build.environment: <string>,
 *       experienceCloud.org: <string>,
 *       property.id: <string>,
 *       rules.url: <string>,
 *     },
 *     ACPExtensionEventSource: 'com.adobe.eventsource.responsecontent'
 *     ACPExtensionEventType: 'com.adobe.eventtype.configuration'
 *     ACPExtensionEventName: <string>,
 *     ACPExtensionEventNumber: <integer>,
 *     ACPExtensionEventUniqueIdentifier: <string>,
 *   },
 *   type: 'generic'
 *   vendor: <string>,
 *   annotations: <array>,
 *   clientId: <string>,
 *   timestamp: <number>,
 *   uuid: <string>,
 * }
 * ```
 *
 * @namespace configuration
 */

/**
 * Paths for the keys on a Configuration Event
 *
 * @enum {string}
 */
const path = {
  /** An object with custom data describing the event.<br />Path is `payload`. */
  payload: 'payload',

  /** The full list of current configuration values.<br />Path is `payload.ACPExtensionEventData`. */
  eventData: 'payload.ACPExtensionEventData',

  /** In the Launch UI, the type of environment this configuration was generated for.<br />Path is `payload.ACPExtensionEventData."build.environment"`. */
  buildEnvironment: 'payload.ACPExtensionEventData."build.environment"',

  /** The IMS Org that the mobile app's config was created by.<br />Path is `payload.ACPExtensionEventData."experienceCloud.org"`. */
  experienceCloudOrg: 'payload.ACPExtensionEventData."experienceCloud.org"',

  /** The ID of the property inside launch.<br />Path is `payload.ACPExtensionEventData."property.id"`. */
  launchPropertyId: 'payload.ACPExtensionEventData."property.id"',

  /** The URL to download the rules configuration for the property.<br />Path is `payload.ACPExtensionEventData."rules.url"`. */
  rulesUrl: 'payload.ACPExtensionEventData."rules.url"',

  /** The event source.<br />Path is `payload.ACPExtensionEventSource`. */
  eventSource: 'payload.ACPExtensionEventSource',

  /** The event type.<br />Path is `payload.ACPExtensionEventType`. */
  eventType: 'payload.ACPExtensionEventType',

  /** The name of the event.<br />Path is `payload.ACPExtensionEventName`. */
  eventName: 'payload.ACPExtensionEventName',

  /** The event number generated by the SDK.<br />Path is `payload.ACPExtensionEventNumber`. */
  sdkEventNumber: 'payload.ACPExtensionEventNumber',

  /** The unique event id.<br />Path is `payload.ACPExtensionEventUniqueIdentifier`. */
  eventId: 'payload.ACPExtensionEventUniqueIdentifier',

  /** The type of event.<br />Path is `type`. */
  rootType: 'type',

  /** The vendor of the plugin that sent the event.<br />Path is `vendor`. */
  vendor: 'vendor',

  /** Array of Annotation objects.<br />Path is `annotations`. */
  annotations: 'annotations',

  /** A unique id that differentiates clients from one another.<br />Path is `clientId`. */
  clientId: 'clientId',

  /** When the event occurred.<br />Path is `timestamp`. */
  timestamp: 'timestamp',

  /** Uniquely identifies each event.<br />Path is `uuid`. */
  rootId: 'uuid'
};

/**
 * Describes the number of parents this object has based off schema references. When checking for matches for example, we want to
 * use a schema that is more specific over a more generic schema
 *
 * @constant
 */
const parentDepth = 2;

/**
 * A label that can be used when describing this object
 */
const label = 'Configuration Event';

/**
 * A grouping for this object
 */
const group = 'event';

/**
 * The value for `eventSource` for a Configuration Event.
 *
 * Path is `payload,ACPExtensionEventSource`.
 *
 * @constant
 */
const EVENT_SOURCE = 'com.adobe.eventsource.responsecontent';

/**
 * The value for `eventType` for a Configuration Event.
 *
 * Path is `payload,ACPExtensionEventType`.
 *
 * @constant
 */
const EVENT_TYPE = 'com.adobe.eventtype.configuration';

/**
 * The value for `rootType` for a Configuration Event.
 *
 * Path is `type`.
 *
 * @constant
 */
const ROOT_TYPE = 'generic';

/**
 * Retrieves a value from the object. You can provide either a path or an alias.
 *
 * @function
 * @param path or alias
 * @param {*} data Data to search
 * @returns {*}
 */
const get = R.curry((alias, data) => kit.search(path[alias] || alias, data));

/**
 * Returns the `eventData` from the Configuration Event.
 * This is the the full list of current configuration values.
 *
 * Path is `payload,ACPExtensionEventData`.
 *
 * @function
 * @param {object} source The Configuration Event instance
 * @returns {object}
 */
const getEventData = kit.search(path.eventData);

/**
 * Returns the data using the specified path from the eventData
 * of the Configuration Event.
 *
 * @function
 * @param {...string} path key in object
 * @param {object} source The Configuration Event instance
 * @returns {*}
 */
const getEventDataKey = kit.curry(
  (searchPath, source) => kit.search(`${path.eventData}.${searchPath}`, source)
);

/**
 * Returns the `buildEnvironment` from the Configuration Event.
 * This is the in the Launch UI, the type of environment this configuration was generated for.
 *
 * Path is `payload,ACPExtensionEventData,build.environment`.
 *
 * @function
 * @param {object} source The Configuration Event instance
 * @returns {string}
 */
const getBuildEnvironment = kit.search(path.buildEnvironment);

/**
 * Returns the `experienceCloudOrg` from the Configuration Event.
 * This is the the IMS Org that the mobile app's config was created by.
 *
 * Path is `payload,ACPExtensionEventData,experienceCloud.org`.
 *
 * @function
 * @param {object} source The Configuration Event instance
 * @returns {string}
 */
const getExperienceCloudOrg = kit.search(path.experienceCloudOrg);

/**
 * Returns the `launchPropertyId` from the Configuration Event.
 * This is the the ID of the property inside launch.
 *
 * Path is `payload,ACPExtensionEventData,property.id`.
 *
 * @function
 * @param {object} source The Configuration Event instance
 * @returns {string}
 */
const getLaunchPropertyId = kit.search(path.launchPropertyId);

/**
 * Returns the `rulesUrl` from the Configuration Event.
 * This is the the URL to download the rules configuration for the property.
 *
 * Path is `payload,ACPExtensionEventData,rules.url`.
 *
 * @function
 * @param {object} source The Configuration Event instance
 * @returns {string}
 */
const getRulesUrl = kit.search(path.rulesUrl);

/**
 * Matcher can be used to find matching Configuration Event objects.
 *
 * @see kit.match
 * @constant
 */
const matcher = kit.combineAll([
  'payload.ACPExtensionEventSource==`com.adobe.eventsource.responsecontent`',
  'payload.ACPExtensionEventType==`com.adobe.eventtype.configuration`',
  'timestamp'
]);

/**
 * Tests the provided source against the matcher to see if it's Configuration Event event.
 *
 * @function
 * @param {object} source The Configuration Event instance
 * @returns {boolean}
 * @see kit.isMatch
 */
const isMatch = (source) => kit.isMatch(matcher, source);
/**
 * Generates a Configuration Event with the const values set.
 * Can be useful in testing.
 * Can provide additional data by providing a flat object of paths and values.
 *
 * @function
 * @param {...Function} input Overrides
 * @returns {object}
 */
const make = (input) => kit.expandWithPaths(path, {
  eventSource: 'com.adobe.eventsource.responsecontent',
  eventType: 'com.adobe.eventtype.configuration',
  rootType: 'generic',
  ...input
});

/**
 * Generates a Configuration Event with some default values set.
 * Can be useful in testing.
 * Can override defaults and provide additional data by providing a flat object
 * of paths and values.
 *
 * @function
 * @param {...Function} input Overrides
 * @returns {object}
 */
const mock = (input) => kit.expandWithPaths(path, {
  buildEnvironment: 'dev',
  experienceCloudOrg: 'abc@AdobeOrg',
  launchPropertyId: 'abcd1234',
  rulesUrl: 'http://assets.adobedtm.com/abc/abcdefg-development-rules.zip',
  eventSource: 'com.adobe.eventsource.responsecontent',
  eventType: 'com.adobe.eventtype.configuration',
  rootType: 'generic',
  vendor: 'com.adobe.mobile.sdk',
  clientId: 'appleABC',
  timestamp: Date.parse('12 Jan 2020 07:23:17 GMT'),
  rootId: '123',
  ...input
});

/* ADD CUSTOM CONTENT BELOW */

// additional exports should be added here:
const customExports = {};

/* END CUSTOM CONTENT */
/* The content below is autogenerated. Do not make any changes */

export default {
  path,
  mock,
  make,
  schema,
  get,
  ...customExports,
  getEventData,
  getEventDataKey,
  getBuildEnvironment,
  getExperienceCloudOrg,
  getLaunchPropertyId,
  getRulesUrl,
  isMatch,
  matcher,
  EVENT_SOURCE,
  EVENT_TYPE,
  ROOT_TYPE,
  label,
  group,
  parentDepth
};
