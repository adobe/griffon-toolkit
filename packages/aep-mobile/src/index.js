// @create-index

export { default as clientInfoAndroid } from './clientInfoAndroid.js';
export { default as clientInfoIOS } from './clientInfoIOS.js';
export { default as configUpdateCommand } from './configUpdateCommand.js';
export { default as fakeEventCommand } from './fakeEventCommand.js';
export { default as genericPlaces } from './genericPlaces.js';
export { default as logForwardCommand } from './logForwardCommand.js';
export { default as mobileEvent } from './mobileEvent.js';
export { default as placesEntry } from './placesEntry.js';
export { default as placesExit } from './placesExit.js';
export { default as receivePlaces } from './receivePlaces.js';
export { default as requestPlaces } from './requestPlaces.js';
export { default as screenshotCaptureCommand } from './screenshotCaptureCommand.js';
export { default as startEventForwardingCommand } from './startEventForwardingCommand.js';
export { default as startPlacesMonitor } from './startPlacesMonitor.js';

