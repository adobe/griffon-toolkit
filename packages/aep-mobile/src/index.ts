// created from 'create-ts-index'

export * from './types';
export * from './analyticsResponse';
export * from './clientInfoAndroid';
export * from './clientInfoIOS';
export * from './configUpdateCommand';
export * from './configuration';
export * from './configurationUpdate';
export * from './edgeErrorResponse';
export * from './edgeRequest';
export * from './edgeResponse';
export * from './fakeEventCommand';
export * from './genericPlaces';
export * from './genericTrack';
export * from './identityResponse';
export * from './lifecycleStart';
export * from './locationAuthStatus';
export * from './logForwardCommand';
export * from './mediaCreateTrackerRequest';
export * from './mediaSessionCreated';
export * from './mobileEvent';
export * from './placesEntry';
export * from './placesExit';
export * from './pushTrackingRequest';
export * from './receivePlaces';
export * from './requestPlaces';
export * from './screenshotCaptureCommand';
export * from './setPushIdentifier';
export * from './sharedState';
export * from './sharedStateConfig';
export * from './sharedStateVersions';
export * from './startEventForwardingCommand';
export * from './startPlacesMonitor';
export * from './trackAction';
export * from './trackMedia';
export * from './trackState';
