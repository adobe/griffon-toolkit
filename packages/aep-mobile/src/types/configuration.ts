import { ACPExtensionEventSource, ACPExtensionEventType, MobileEvent } from "./mobileEvent";

export interface ConfigurationEventPayload {
    ACPExtensionEventData: {
        /** The full list of current configuration values */
        'build.environment': string;
        /** The IMS Org that the mobile app's config was created by */
        'experienceCloud.org': string;
        /** The ID of the property inside launch */
        'property.id': string;
        /** The URL to download the rules configuration for the property */
        'rules.url': string;
    };
    ACPExtensionEventSource: ACPExtensionEventSource.RESPONSECONTENT;
    ACPExtensionEventType: ACPExtensionEventType.CONFIGURATION;
}

/** Configuration Event */
export interface ConfigurationEvent extends MobileEvent {
    payload: ConfigurationEventPayload;
}