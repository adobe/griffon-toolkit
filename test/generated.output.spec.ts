/*
Copyright 2020 Adobe. All rights reserved.
This file is licensed to you under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License. You may obtain a copy
of the License at http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software distributed under
the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
OF ANY KIND, either express or implied. See the License for the specific language
governing permissions and limitations under the License.
*/

// import jmespath from 'jmespath';
import entry from '../packages/aep-mobile/src/placesEntry';
import { PlacesEntry } from '../packages/aep-mobile';
import aep from '../packages/aep-mobile/src/mobileEvent';
import root from '../packages/common/src/event';
import configuration from '../packages/aep-mobile/src/configuration';
import iosClient from '../packages/aep-mobile/src/clientInfoIOS';

describe('Test Auto Generated Output', () => {
  it('exports paths', () => {
    expect(Object.keys(root.path).length).toBe(6);
  });
  it('exports inherited paths', () => {
    expect(Object.keys(aep.path).length).toBe(13);
  });
  it('exports paths from object refs', () => {
    expect(Object.keys(entry.path).length).toBe(29);
  });
  it('creates a make command', () => {
    expect(root.make({})).toEqual({});

    const entryEvent = entry.make(null) as PlacesEntry;
    expect(Object.keys(entryEvent.payload).length).toBe(3);
    expect(entry.getRegionEventType(entryEvent)).toBe('entry');
  });
  it('matches against matchers', () => {
    const events = [
      entry.mock({}),
      aep.mock({}),
      root.mock({})
    ];
    expect(entry.isMatch(events[0])).toBe(true);
    expect(entry.isMatch(events[1])).toBe(false);
  });
  it('exports constants', () => {
    expect(entry.REGION_EVENT_TYPE).toBe('entry');
  });
  it('exports getters', () => {
    const mock = root.mock({});
    expect(root.getClientId(mock)).toBe('appleABC');
    expect(root.getRootType(mock)).toBe('generic');
  });
  it('generates a generic get', () => {
    const entryEvent = entry.mock({
      city: 'Appleton'
    });
    expect(entry.get('city', entryEvent)).toBe('Appleton');
    expect(entry.get(entry.path.city, entryEvent)).toBe('Appleton');
  });
  it('exports the parent depth', () => {
    expect(root.parentDepth).toBe(0);
    expect(aep.parentDepth).toBe(1);
    expect(entry.parentDepth).toBe(3);
  });
  it('exports get from object functions', () => {
    const mock = root.mock({
      [root.path.payload]: {
        size: 'L',
        dimensions: { width: 800, height: 600 }
      }
    });
    expect(root.getPayloadKey('size', mock)).toBe('L');
    expect(root.getPayloadKey('dimensions.width', mock)).toBe(800);
  });
  it('handles keys with spaces in them', () => {
    // some schemas have spaces in the key names. This test validates that those
    // schemas work correctly
    const mock = iosClient.mock({
      [iosClient.path.os]: 'iOS 13'
    });

    expect(iosClient.isMatch(mock)).toBe(true);
    expect(iosClient.getPlatform(mock)).toBe('iOS');
    expect(iosClient.getOs(mock)).toBe('iOS 13');
  });
  it('handles keys with dots in them', () => {
    const mock = configuration.mock({
      launchPropertyId: 'abcd',
      eventData: { box: true }
    });
    expect(configuration.getLaunchPropertyId(mock)).toBe('abcd');
    expect(configuration.getEventData(mock)['property.id']).toBe('abcd');
  });
});
