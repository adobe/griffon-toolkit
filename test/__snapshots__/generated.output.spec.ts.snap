// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Auto Generated Output can provide custom matchers 1`] = `"payload.ACPExtensionEventData.type==\`personalization:decisions\` && (payload.ACPExtensionEventSource!=\`com.adobe.eventSource.errorResponseContent\` && payload.ACPExtensionEventSource!=\`com.adobe.eventSource.requestContent\`) && (payload.ACPExtensionEventType==\`com.adobe.eventType.edge\` || payload.ACPExtensionEventType==\`com.adobe.eventtype.edge\`) && timestamp && payload.ACPExtensionEventData.payload[0].placement.id"`;
